launch/pipeline/utils.py

```diff
@@ -21,14 +43,108 @@ def make_service(
         service=service,
         runtime=runtime,
         deployment=deployment,
+        env_params=env_params,
         **kwargs,
     )
 
 
 def make_sequential_pipeline(
     items: List[SingleServiceDescription],
+    runtime: Runtime,
+    deployment: Deployment,
+    env_params: Dict[str, str],
 ) -> SequentialPipelineDescription:
     """
     Create a structure that describes a sequential pipeline.
     """
-    return SequentialPipelineDescription(items=items)
+    return SequentialPipelineDescription(
+        items=items,
+        runtime=runtime,
+        deployment=deployment,
+        env_params=env_params,
+    )
+
+
+def deploy_service(
```

Comment:
I think this function should go in the main client file. IMO it's a bit surprising to have deployment done in two different places.

Reply:
Yeah it kinda seems like we're creating a parallel API in here. Granted, that was fine for the API exercise that @malcolmgreaves helped with, but I think at implementation time, we may want to keep compatibility and merge the concepts together.